lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp

```diff
@@ -0,0 +1,666 @@
+                        HdRetainedTypedSampledDataSource<VtVec3fArray>::New(
+                            VtVec3fArray{
+                                displayColor, 
+                            }))
+                    .SetIndices(
+                        HdRetainedTypedSampledDataSource<VtIntArray>::New(
+                                 vertexColorArray
+                        )
+                    )
+                    .SetInterpolation(
+                        HdPrimvarSchema::BuildInterpolationDataSource(
+                            HdPrimvarSchemaTokens->varying))
+                    .SetRole(
+                        HdPrimvarSchema::BuildRoleDataSource(
+                            HdPrimvarSchemaTokens->color))//vertex color
+                    .Build()
+            );
+
+        //Apply the size of the prim as a scale matrix
+        GfMatrix4d transform;
+        transform.SetIdentity();
+        transform.SetScale(scale);
+
+        //Create the primitive
+        HdRetainedSceneIndex::AddedPrimEntry addedPrim;
+        addedPrim.primPath   = primPath;
+        addedPrim.primType   = HdPrimTypeTokens->mesh;
+        addedPrim.dataSource = HdRetainedContainerDataSource::New(
+            //Create a matrix
+            HdXformSchemaTokens->xform,
+            HdXformSchema::Builder()
+                    .SetMatrix(HdRetainedTypedSampledDataSource<GfMatrix4d>::New(
+                    transform)).Build(),
+
+            //create a mesh
+            HdMeshSchemaTokens->mesh,
+            meshDs,
+            HdPrimvarsSchemaTokens->primvars,
+            primvarsDs
+        );
+
+        //Add the prim in the retained scene index
+        retainedSceneIndex->AddPrims({addedPrim});
+    }
+
+    //Callback when an attribute of the maya node changes
+    void attributeChangedCallback(MNodeMessage::AttributeMessage msg, MPlug& plug, MPlug & otherPlug, void* footPrintData)
+    {
+        if (! footPrintData){
+            return;
+        }
+
+        MhFootPrint* footPrint = reinterpret_cast<MhFootPrint*>(footPrintData);
+
+        MPlug parentPlug = plug.parent();
+        if (
+            (plug       == MhFootPrint::mSize)  ||
+            (parentPlug == MhFootPrint::mColor) ||
+            (plug       == MhFootPrint::mColor)
+           ){
+                footPrint->UpdateFootPrintPrims();
+        }
+    }
+
+    //Get the color attribute value which is a double3
+    void GetDouble3AttributeValue(double3& outVal, const MObject& node, const MObject& attr)
+    {
+        MPlug plug(node, attr);
+        if (plug.isNull()){
+            return;
+        }
+
+        MObject oDouble3;
+        plug.getValue(oDouble3);
+
+        MFnNumericData fnData(oDouble3);
+        fnData.getData( outVal[0], outVal[1], outVal[2] );
+    }
+}//end of anonymous namespace
+
+//Static variables init
+int     MhFootPrint::_counter = 0;
+MObject MhFootPrint::mSize;
+MObject MhFootPrint::mColor;
+MTypeId MhFootPrint::id( 0x90573 );
```

Comment:
I trust you registered this plugin ID wherever it now needs to be registered.

Reply:
I've fixed that now.

Reply:
Thanks, I've added one PR for that in maya.